account.rs/src/wasm/src/account.rs

```diff
@@ -0,0 +1,345 @@
+        let mut account = Self::new(&burner_seed_string)?;
+        account.burner_seed = Some(burner_seed_string);
+        Ok(account)
+    }
+
+    #[wasm_bindgen(js_name = fromPrivateKey)]
+    pub fn from_private_key(
+        private_key: Vec<u8>,
+        encryption_private_key: Vec<u8>,
+        aes_secret: Vec<u8>,
+    ) -> Self {
+        let public_key = Account::generate_shielded_public_key(private_key.to_vec());
+        let encryption_public_key =
+            Account::create_encryption_public_key(encryption_private_key.to_vec());
+
+        Account {
+            // solana_public_key,
+            private_key,
+            public_key,
+
+            encryption_public_key,
+            encryption_private_key,
+
+            aes_secret,
+            hashing_secret: None,
+
+            burner_seed: None,
+
+            prefix_counter: BigUint::from(0u32),
+        }
+    }
+
+    #[wasm_bindgen(js_name = generateShieldedPrivateKey)]
+    pub fn generate_shielded_private_key(seed: &str) -> Result<Vec<u8>, AccountError> {
+        let private_key_seed = format!("{}shielded", seed);
+        let blake_hash = blake2_string(private_key_seed, ACCOUNT_HASH_LENGTH);
+        let trimmed_hash =
+            BigUint::from_bytes_be(&blake_hash[1..ACCOUNT_HASH_LENGTH]).to_bytes_be();
+        poseidon_hash(vec![trimmed_hash]).map_err(AccountError)
+    }
+
+    #[wasm_bindgen(js_name = generateShieldedPublicKey)]
+    pub fn generate_shielded_public_key(private_key: Vec<u8>) -> Vec<u8> {
+        poseidon_hash(vec![private_key]).unwrap()
+    }
+
+    #[wasm_bindgen(js_name = createEncryptionPrivateKey)]
+    pub fn create_encryption_private_key(seed: &str) -> Vec<u8> {
+        let encryption_seed = format!("{}encryption", seed);
+        let blake_hash = blake2_string(encryption_seed, ACCOUNT_HASH_LENGTH);
+        let secret_key_bytes: [u8; ACCOUNT_HASH_LENGTH] = blake_hash.as_slice().try_into().unwrap();
+        let secret_key = SecretKey::from(secret_key_bytes);
+        secret_key.as_bytes().to_vec()
+    }
+
+    #[wasm_bindgen(js_name = createEncryptionPublicKey)]
+    pub fn create_encryption_public_key(encryption_private_key: Vec<u8>) -> Vec<u8> {
+        let secret_key_bytes: [u8; ACCOUNT_HASH_LENGTH] =
+            encryption_private_key.try_into().unwrap();
+        let secret_key = SecretKey::from(secret_key_bytes);
+        secret_key.public_key().as_bytes().to_vec()
+    }
+
+    #[wasm_bindgen(js_name = generateSecret)]
+    pub fn generate_secret(seed: &str, domain: &str) -> Vec<u8> {
+        let input = format!("{}{}", seed, domain);
+        blake2_string(input, ACCOUNT_HASH_LENGTH)
+    }
+
+    pub fn sign(&self, commitment: String, merkle_path: u32) -> Vec<u8> {
+        let inputs: Vec<Vec<u8>> = vec![
+            self.private_key.to_vec(),
+            commitment.into_bytes(),
+            merkle_path.to_be_bytes().to_vec(),
+        ];
+        poseidon_hash(inputs).unwrap()
```

Comment:
Declaring a vector and using `to_vec()` seems a bit suboptimal for me (you are allocating memory and casting types). You could just construct a slice of slice bytes.

```suggestion
        let inputs: &[&[u8]] = &[
            self.private_key.as_slice(),
            commitment.into_bytes().as_slice(),
            merkle_path.to_be_bytes().as_slice(),
        ];
        poseidon_hash(inputs).unwrap()
```

That should be more efficient.

If for some reason borrow checker complains about constructing the `inputs` variable this way, you could try to just construct such slice directly in the `poseidon_hash` call:

```suggestion
        poseidon_hash(&[
            self.private_key.as_slice(),
            commitment.into_bytes().as_slice(),
            merkle_path.to_be_bytes().as_slice(),
        ]).unwrap()
```